pygac/l1c_factory.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+
+# Copyright (c) 2014, 2019 Pygac Developers
+
+# Author(s):
+
+#   Carlos Horn <carlos.horn@external.eumetsat.int>
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Factory for l1c files.
+
+"""
+import datetime
+import logging
+
+from pygac.gac_klm import GACKLMReader
+from pygac.gac_pod import GACPODReader
+from pygac.lac_klm import LACKLMReader
+from pygac.lac_pod import LACPODReader
+from pygac.utils import file_opener
+
+
+LOG = logging.getLogger(__name__)
+
+
+class L1cBuilder(object):
```

Comment:
I don't see the point of having this as a separate class. Why not just add a eg `process` function to the `Reader` class ? This would also allow to factorize the `main_klm` and `main_pod` functions to just call the reader's `process` function.

Reply:
Things like the `Reader._prepare_channels` could have gone into it, but you are right, adding a process function to the `Reader` makes it simpler.